src/lib/y2storage/proposal/autoinst_space_maker.rb

```diff
@@ -131,6 +158,76 @@ def register_invalid_use_value(drive_spec)
           issues_list.add(:missing_value, drive_spec, :use)
         end
       end
+
+      # Return a map of reused partitions
+      #
+      # Calculates which partitions are meant to be reused and, as a consequence, should not
+      # be deleted.
+      #
+      # @param devicegraph     [Devicegraph]               devicegraph
+      # @param planned_devices [Array<Planned::Partition>] set of partitions
+      # @return [Hash<String,Array<String>>] disk name to list of reused partitions map
+      def reused_partitions_by_disk(devicegraph, planned_devices)
+        find_reused_partitions(devicegraph, planned_devices).each_with_object({}) do |part, map|
+          disk_name = part.disk.name
+          map[disk_name] ||= []
+          map[disk_name] << part.name
+        end
+      end
+
+      # Determine which partitions will be reused
+      #
+      # @param devicegraph     [Devicegraph]               devicegraph
+      # @param planned_devices [Array<Planned::Partition>] set of partitions
+      # @return [Hash<String,Array<String>>] disk name to list of reused partitions map
+      def find_reused_partitions(devicegraph, planned_devices)
+        reused_devices = planned_devices.select(&:reuse).each_with_object([]) do |device, all|
+          case device
+          when Y2Storage::Planned::Partition
+            all << devicegraph.partitions.find { |p| device.reuse == p.name }
+          when Y2Storage::Planned::LvmVg
+            vg = devicegraph.lvm_vgs.find { |v| File.join("/dev", device.reuse) == v.name }
+            all.concat(vg.lvm_pvs.map(&:blk_device))
```

Comment:
Method `LvmPv#blk_device` could return the encryption device instead of the blk device. There is no problem if the device hosting the PV is a partition due to you are searching then by partitions in the ancestors list. But there is a possible corner case: a PV over an encrypted disk. Is it necessary to take into account that scenario? Probably not because you are only working with partitions here, isn't it?

Reply:
Thanks for the info. Yes, I am working with partitions. If I can reach the partition through the "#ancestors" method, it will be ok. If not, I guess I should handle it.